src/server.cc

```diff
@@ -54,3 +60,48 @@ bool Server::handle_accept(Connection::pointer new_connection,
   start_accept();
   return true;
 }
+
+bool Server::getServerInfo(const char* file_name, ServerInfo* info) {
+  NginxConfigParser parser;
+  NginxConfig config;
+  if (!parser.Parse(file_name, &config)) {
+    return false;
+  }
+  
+  for(unsigned i = 0; i < config.statements_.size(); i++) {
+    std::string key = config.statements_[i]->tokens_[0];
+    if(key == "port") {
+      info->port = std::stoi(config.statements_[i]->tokens_[1]);
+    } else if (key == "path") {
+      if(config.statements_[i]->tokens_.size() != 3
+          || !config.statements_[i]->child_block_) {
+        return false;
+      }
+      // determine handler type
+      if(config.statements_[i]->tokens_[2] == "StaticFileHandler") {
```

Comment:
Since we haven't finished API implementation, these will change once we've finished, right?